### PR TITLE
Update operator --set help text #20204

### PR DESCRIPTION
**Pull request by blhagadorn**
This is a repeat of https://github.com/istio/istio/pull/20204 without all of the commits I didn't make addresses issue https://github.com/istio/istio/issues/19999 and functionally chages the Help string as well as adds `Example:` to cobraCommands. 

[ ] Configuration Infrastructure
[ x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ x] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: operator/cmd/mesh/manifest-apply.go [hunk 1]
```diff
@@ -63,7 +63,11 @@ func manifestApplyCmd(rootArgs *rootArgs, maArgs *manifestApplyArgs) *cobra.Comm
 		Use:   "apply",
 		Short: "Generates and applies an Istio install manifest.",
 		Long:  "The apply subcommand generates an Istio install manifest and applies it to a cluster.",
-		Args:  cobra.ExactArgs(0),
+		Example: "istioctl manifest apply  # installs the default profile on the current Kubernetes cluster context\n" +
+			"istioctl manifest apply --set values.global.mtls.enabled=true --set values.global.controlPlaneSecurityEnabled=true\n" +
+			"istioctl manifest apply --set profile=demo\n" +
+			"istioctl manifest apply --set installPackagePath=~/istio-releases/istio-1.4.3/install/kubernetes/operator/charts",
```
Comment (howardjohn) [inline operator/cmd/mesh/manifest-apply.go]: @ostromart I didn't think we expand `~`, am I wrong?

File: operator/cmd/mesh/root.go [hunk 2]
```diff
@@ -24,9 +24,10 @@ import (
 )
 
 const (
-	SetFlagHelpStr = `Set a value in IstioOperator CustomResource. e.g. --set policy.enabled=true.
-Overrides the corresponding path value in the selected profile or passed through IstioOperator CR
-customization file`
+	SetFlagHelpStr = `Override an IstioOperator value, e.g. to choose a profile
+(--set profile=demo), enable or disable components (--set policy.enabled=true), or override Istio
```
Comment (howardjohn) [inline operator/cmd/mesh/root.go]: I think this is `-- components.policy.enabled=true` now?

File: operator/cmd/mesh/root.go [hunk 3]
```diff
@@ -24,9 +24,10 @@ import (
 )
 
 const (
-	SetFlagHelpStr = `Set a value in IstioOperator CustomResource. e.g. --set policy.enabled=true.
-Overrides the corresponding path value in the selected profile or passed through IstioOperator CR
-customization file`
+	SetFlagHelpStr = `Override an IstioOperator value, e.g. to choose a profile
+(--set profile=demo), enable or disable components (--set policy.enabled=true), or override Istio 
+settings(--set values.global.mtls.enabled=true).  See documentation for more info:
```
Comment (howardjohn) [inline operator/cmd/mesh/root.go]:
```suggestion
settings (--set values.global.mtls.enabled=true). See documentation for more info: 
```

**Review by howardjohn — COMMENTED** [review]
nice!

/ok-to-test

File: operator/cmd/mesh/manifest-apply.go (hunk 1 above)
Comment (blhagadorn) [reply operator/cmd/mesh/manifest-apply.go]: Took that from this, https://istio.io/docs/setup/install/istioctl/#install-from-external-charts, happy to alter it and throw in an absolute directory (and update istio.io).

File: operator/cmd/mesh/root.go (hunk 2 above)
Comment (blhagadorn) [reply operator/cmd/mesh/root.go]: Yes you're right, good catch. Will change

**Review by howardjohn — APPROVED** [review]

**Pull request merged**